internals/overlord/cmdstate/request.go

```diff
@@ -117,12 +117,15 @@ func Exec(st *state.State, args *ExecArgs) (*state.Task, ExecMetadata, error) {
 		environment["LANG"] = "C.UTF-8"
 	}
 
-	// Set default working directory to $HOME, or / if $HOME not set.
+	// Set default working directory to $HOME, or / if $HOME not set or doesn't exist.
 	workingDir := args.WorkingDir
 	if workingDir == "" {
 		workingDir = environment["HOME"]
 		if workingDir == "" {
 			workingDir = "/"
+		} else if !osutil.IsDir(workingDir) {
+			logger.Debugf("HOME dir %q does not exist, using /", workingDir)
+			workingDir = "/"
```

Comment:
if we move all this logic into a separate function, we could use early returns and avoid all the nesting

Reply:
I've moved this to a separate function as the stat / error handling logic is now quite a lot.